### ISSUE TITLE
Import as pure

### ISSUE THREAD

**Issue by mlutze** [flix/flix#2498]
Whenever we use a pure Java function, we have to annotate the expression using it with `as & Pure`.
This can be dangerous, since a cast could capture some other impurity in the expression. For example:

```scala
def linesInMyFile(): Int & Impure = ???

def linesInMyFilePlusHeader(): Int = {
  import matts.java.lib.Math.succ(Int32);
  succ(linesInMyFile()) as & Pure // casting because succ is actually pure
}
```

We could partially avoid this problem by saying that a function is pure when we import it
```scala
def linesInMyFilePlusHeader(): Int = {
  import matts.java.lib.Math.succ(Int32) as & Pure;
  succ(linesInMyFile()) // No need to cast. I see that I have to mark this Impure because of linesInMyFile()
}
```

Alternative syntax
```scala
import Pure matts.java.lib.Math.succ(Int32);
```

**Comment by magnus-madsen** [comment]
I suggest the syntax:

```scala
import java.lang.Foo.bar(): Int32 & Pure
```

to mirror the syntax of defs.

**Comment by magnus-madsen** [comment]
This is being addressed in #2512.
